vpr/src/route/connection_router.cpp

```diff
@@ -210,6 +222,20 @@ t_heap* ConnectionRouter<Heap>::timing_driven_route_connection_from_heap(int sin
 
         //Have we found the target?
         if (inode == sink_node) {
+            // If we're running RCV, the path will be stored in the path_data->path_rr vector
+            // This is then placed into the traceback so that the correct path is returned
+            // TODO: This can be eliminated by modifying the actual traceback function in route_timing
+            if (cost_params.delay_budget && cost_params.delay_budget->routing_budgets_algorithm == YOYO) {
```

Comment:
Shouldn't this be `if(cheapest->path_data)` rather than `(cost_params.delay_budget && cost_params.delay_budget->routing_budgets_algorithm == YOYO)` ? E.g. any time path_data is in use, do the following?

Reply:
You're right, but ideally these conditions would be identical as path_data should only be initialized if RCV is enabled, and vice versa. I can change this because it might be slightly faster to compute and read.